basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java

```diff
@@ -935,10 +935,83 @@ public static void addGlobalData(Site site, Properties props, Properties custom,
 				M_log.debug("Launching with SHA256 Signing");
 			}
 		}
+		
+		//substitute resources params and add the rest
+		if(custom.getProperty(BasicLTIConstants.RESOURCE_LINK_ID) != null){//TODO if allowReviewService prop y future prop for TII
+			addTIIproperties(ltiProps, custom);
```

Comment:
I don't see the point of this code anyway. All of this data is contained in the launch. If turnitin can't process it for some reason that sounds more like their problem. But if you can't work around them, then this code should be a lot more generic. (Maybe a list of custom properties to "promote" that could be set via configuration. I wouldn't merge with this very specific code in here.